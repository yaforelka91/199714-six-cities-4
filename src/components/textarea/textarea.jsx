import React from 'react';
import {textareaTypes} from '../../types/types';

const Textarea = ({currentMessage, onTextareaChange}) => {
  return (
    <textarea
      className="reviews__textarea form__textarea"
      id="review" name="review"
      value={currentMessage}
      placeholder="Tell how was your stay, what you like and what can be improved"
      onChange={onTextareaChange}
    />
  );
};

Textarea.defaultProps = {
  currentMessage: ``,
};

Textarea.propTypes = textareaTypes;

export {Textarea};
export default React.memo(Textarea);
