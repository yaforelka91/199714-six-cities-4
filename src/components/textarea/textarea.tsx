import * as React from 'react';

type Props = {
  currentMessage: string;
  maxLength: number;
  onTextareaChange: () => void;
};

const Textarea: React.FC<Props> = (props: Props) => {
  const {
    currentMessage = ``,
    maxLength = 100,
    onTextareaChange
  } = props;
  return (
    <textarea
      className="reviews__textarea form__textarea"
      id="review" name="review"
      value={currentMessage}
      maxLength={maxLength}
      placeholder="Tell how was your stay, what you like and what can be improved"
      onChange={onTextareaChange}
    />
  );
};

export {Textarea};
export default React.memo(Textarea);
