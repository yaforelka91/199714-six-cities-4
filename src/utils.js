import {createSelector} from 'reselect';
import {SortType, CardView, MONTHS_LIST} from './const.js';

export const extend = (object1, object2) => {
  return Object.assign({}, object1, object2);
};

export const sortOffers = (sortType, offers) => {
  switch (sortType) {
    case SortType.POPULAR:
      return offers.slice();
    case SortType.TO_HIGH:
      return offers.slice().sort((prevOffer, nextOffer) => prevOffer.price - nextOffer.price);
    case SortType.TO_LOW:
      return offers.slice().sort((prevOffer, nextOffer) => nextOffer.price - prevOffer.price);
    case SortType.TOP_RATED:
      return offers.slice().sort((prevOffer, nextOffer) => nextOffer.rating - prevOffer.rating);
    default:
      return offers;
  }
};

export const capitalize = (word) => {
  return word.charAt(0).toUpperCase() + word.slice(1);
};

export const getFormattedDate = (dateStr) => {
  const date = new Date(dateStr);
  return `${MONTHS_LIST[date.getMonth()]} ${date.getFullYear()}`;
};

export const getSortedOffers = createSelector(
    (state) => {
      return state.sortType;
    },
    (state) => {
      return state.offers;
    },
    (activeSorting, offers) => {
      return sortOffers(activeSorting, offers);
    }
);

export const getRatingInPercent = (ratingValue) => {
  return Math.round(ratingValue * 100 / 5);
};

export const getArticleClassName = (viewMode) => {
  switch (viewMode) {
    case CardView.CITIES:
      return {
        classNameForArticle: `cities__place-card`,
        classNameForImage: `cities__image-wrapper`,
        classNameForInfo: ``,
      };

    case CardView.NEAR:
      return {
        classNameForArticle: `near-places__card`,
        classNameForImage: `near-places__image-wrapper`,
        classNameForInfo: ``,
      };

    case CardView.FAVORITES:
      return {
        classNameForArticle: `favorites__card`,
        classNameForImage: `favorites__image-wrapper`,
        classNameForInfo: `favorites__card-info`,
      };
  }
  return {
    classNameForArticle: ``,
    classNameForImage: ``,
    classNameForInfo: ``,
  };
};
