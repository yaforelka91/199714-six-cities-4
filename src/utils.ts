import {createSelector} from 'reselect';
import {SortType, CardView, MONTHS_LIST, SortingTypes} from './const';

export const extend = (object1, object2) => Object.assign({}, object1, object2);

export const sortOffers = (sortType: SortingTypes, offers) => {
  switch (sortType) {
    case SortType.POPULAR:
      return offers.slice();
    case SortType.TO_HIGH:
      return offers.slice().sort((prevOffer, nextOffer) => prevOffer.price - nextOffer.price);
    case SortType.TO_LOW:
      return offers.slice().sort((prevOffer, nextOffer) => nextOffer.price - prevOffer.price);
    case SortType.TOP_RATED:
      return offers.slice().sort((prevOffer, nextOffer) => nextOffer.rating - prevOffer.rating);
    default:
      return offers;
  }
};

export const capitalize = (word) => word.charAt(0).toUpperCase() + word.slice(1);

export const getFormattedDate = (dateStr, isFull) => {
  const date = new Date(dateStr);
  return isFull ? dateStr.match(/([^T]+)/)[0] : `${MONTHS_LIST[date.getMonth()]} ${date.getFullYear()}`;
};

export const getSortedOffers = createSelector(
    (state) => state.sortType,
    (state) => state.offers,
    (activeSorting, offers) => sortOffers(activeSorting, offers)
);

export const getRatingInPercent = (ratingValue, isInteger) => {
  return isInteger ? Math.round(ratingValue) * 100 / 5 : Math.round(ratingValue * 100 / 5);
};

export const getArticleClassName = (viewMode: CardView) => {
  switch (viewMode) {
    case CardView.CITIES:
      return {
        classNameForArticle: `cities__place-card`,
        classNameForImage: `cities__image-wrapper`,
        classNameForInfo: ``,
      };

    case CardView.NEAR:
      return {
        classNameForArticle: `near-places__card`,
        classNameForImage: `near-places__image-wrapper`,
        classNameForInfo: ``,
      };

    case CardView.FAVORITES:
      return {
        classNameForArticle: `favorites__card`,
        classNameForImage: `favorites__image-wrapper`,
        classNameForInfo: `favorites__card-info`,
      };
  }

  return {
    classNameForArticle: ``,
    classNameForImage: ``,
    classNameForInfo: ``,
  };
};

export const noOperation = () => null;
