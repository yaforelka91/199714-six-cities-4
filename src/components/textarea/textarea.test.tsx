import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Textarea} from './textarea';
import {noOperation} from '../../utils';

describe(`TextareaSnapshot`, () => {
  it(`should render Textarea`, () => {

    const tree = renderer.create(
        <Textarea
          onTextareaChange={noOperation}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
