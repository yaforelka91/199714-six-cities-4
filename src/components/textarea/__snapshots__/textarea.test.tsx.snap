// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextareaSnapshot should render Textarea 1`] = `
<textarea
  className="reviews__textarea form__textarea"
  id="review"
  maxLength={100}
  name="review"
  onChange={[Function]}
  placeholder="Tell how was your stay, what you like and what can be improved"
  value=""
/>
`;
